foreign-types-macros/src/parse.rs

```diff
@@ -71,6 +77,30 @@ impl Parse for ForeignType {
     }
 }
 
+pub enum ForeignTypeFn {
```

Comment:
I would just parse an `Expr` and let the compiler complain on its own if it's not callable.